src/analysis/processing/qgsalgorithmrandompointsinpolygons.cpp

```diff
@@ -0,0 +1,394 @@
+      baseFeatureProgress += featureProgressStep;
+      feedback->setProgress( baseFeatureProgress );
+      continue;
+    }
+    QgsGeometry polyGeom( polyFeat.geometry() );
+    if ( polyGeom.isEmpty() )
+    {
+      // Increment invalid features count
+      emptyOrNullGeom++;
+      featureCount++;
+      baseFeatureProgress += featureProgressStep;
+      feedback->setProgress( baseFeatureProgress );
+      continue;
+    }
+    if ( mDynamicNumPoints || mDynamicMinDistance || mDynamicMaxAttempts )
+    {
+      expressionContext.setFeature( polyFeat );
+    }
+    // (Re)initialize the local (per polygon) index
+    QgsSpatialIndex localIndex;
+    int localIndexPoints = 0;
+    int pointsAddedForThisFeature = 0;
+    // Get data defined parameters
+    int numberPointsForThisFeature = mNumPoints;
+    if ( mDynamicNumPoints )
+      numberPointsForThisFeature = mNumPointsProperty.valueAsInt( expressionContext, numberPointsForThisFeature );
+    desiredNumberOfPoints += numberPointsForThisFeature;
+    int maxAttemptsForThisFeature = mMaxAttempts;
+    if ( mDynamicMaxAttempts )
+      maxAttemptsForThisFeature = mMaxAttemptsProperty.valueAsInt( expressionContext, maxAttemptsForThisFeature );
+    double minDistanceForThisFeature = mMinDistance;
+    if ( mDynamicMinDistance )
+      minDistanceForThisFeature = mMinDistanceProperty.valueAsDouble( expressionContext, minDistanceForThisFeature );
+    const double pointProgressIncrement = featureProgressStep / ( numberPointsForThisFeature * maxAttemptsForThisFeature );
+    double pointProgress = 0.0;
+    unsigned long fseed = uniformIntDist( mt );
+    // Check if we can avoid using the acceptPoint function
+    if ( ( minDistanceForThisFeature == 0 ) && ( mMinDistanceGlobal == 0 ) )
+    {
+      QVector< QgsPointXY > newPoints = polyGeom.randomPointsInPolygon( numberPointsForThisFeature, fseed );
+      for ( int i = 0; i < newPoints.length(); i++ )
+      {
+        // add the point
+        QgsPointXY pt = newPoints[i];
+        QgsFeature f = QgsFeature( totNPoints );
+        QgsAttributes pAttrs = QgsAttributes();
+        pAttrs.append( totNPoints );
+        if ( mIncludePolygonAttr )
+        {
+          pAttrs.append( polyFeat.attributes() );
+        }
+        f.setAttributes( pAttrs );
+        QgsGeometry newGeom = QgsGeometry::fromPointXY( pt );
+        f.setGeometry( newGeom );
+        sink->addFeature( f, QgsFeatureSink::FastInsert );
+        totNPoints++;
+        pointsAddedForThisFeature++;
+        pointProgress += pointProgressIncrement * ( maxAttemptsForThisFeature );
+      }
+      feedback->setProgress( baseFeatureProgress + pointProgress );
+      continue;
+    }
+    else
+    {
+      // Have to check for minimum distance, provide the acceptPoints function
+      QVector< QgsPointXY > newPoints = polyGeom.randomPointsInPolygon( numberPointsForThisFeature, [ =, &globalIndex, &indexPoints ]( const QgsPointXY & newPoint ) mutable -> bool
```

Comment:
```suggestion
      QVector< QgsPointXY > newPoints = polyGeom.randomPointsInPolygon( numberPointsForThisFeature, [ & ]( const QgsPointXY & newPoint ) -> bool
```
That will capture all available variables as references (and I think you can drop the mutable)